### ISSUE TITLE
Filament <> Sceneform

### ISSUE THREAD

**Issue by michaelvogt** [google-ar/sceneform-android-sdk#329]
The answer to this question might be obvious, but please allow me to answer anyway.

I visited the filament repository again after a while, and saw that the project is pretty active and has quite a bit of stars. Excellent.

I understand that Sceneform uses filament as the renderer behind the scenes (pops up in some logcats here and there). Now I wonder, if I can replace the filament version shipped with Sceneform with an experimental one, in case I'd like to try out a new feature provided there.

Can this be done by just exclude filament from the Sceneform dependency and provide my own one, or is there more to it?

**Comment by romainguy** [comment]
You should be able to just replace the Filament bits in the Sceneform AAR with one compiled yourself. I would recommend rebuilding your custom materials when doing this.

**Comment by michaelvogt** [comment]
With 'rebuilding your custom materials', you mean delete the .sfb files and let them be recreated?

**Comment by tpsiaki** [comment]
You should be able to crack open com.google.sceneform.plugin jar, and replace the matc version(s) in there with your new matc, then rebuild.

**Comment by romainguy** [comment]
@michaelvogt Yes, but you need a new version of the matc shader compiler too. See above
